pkg/hub/controllers/vpnkeyrotation/reconciler.go

```diff
@@ -474,6 +480,25 @@ func (r *Reconciler) updateCertificates(ctx context.Context, rotationVersion int
 	return ctrl.Result{}, false, nil
 }
 
+func (r *Reconciler) areGatewayPodsReady(ctx context.Context, sliceGWName string) bool {
+	podsList := corev1.PodList{}
+	labels := map[string]string{
+		"kubeslice.io/pod-type": "slicegateway",
+		"kubeslice.io/slice-gw": sliceGWName,
+	}
+	listOptions := []client.ListOption{
+		client.MatchingLabels(labels),
+	}
+	err := r.WorkerClient.List(ctx, &podsList, listOptions...)
+	if err != nil {
+		return false
+	}
+	if len(podsList.Items) == 2 {
```

Comment:
@YachikaRalhan will the pod count be always 2?

Reply:
Yes each gateway have two deployments/pods